### ISSUE TITLE
[Docs]: Clarify step ca init --remote-management

### ISSUE THREAD

**Issue by udf2457** [smallstep/certificates#1807]
At present it is hard to see how to correctly transpose [the config shown in the docs](https://smallstep.com/docs/step-ca/cryptographic-protection/#configuring-step-ca-to-use-aws-kms)  to its `step ca init --remote-management` equivalent when setting up a fresh instance.

For example, can `--kms-root=` be left blank ?  The JSON equivalent keys don't make any distinction between root and intermediate.

What would be the command syntax equivalent of the JSON config ? That's what I think ought to be clarified.

Specifically in my use-case, my root is completely offline in HSM, I only have the pub cert available.   The intermediate is in AWS KMS and that's what I'll be using `step ca` against.

**Comment by hslatman** [comment]
Hey @udf2457, initializing `step-ca` with Remote Management enabled doesn't affect how the KMS is configured. It changes how provisioners are managed. Documentation for this can be found [here](https://smallstep.com/docs/step-ca/provisioners/#remote-provisioner-management). So for your use case you should be able to use all of the usual flags to configure the KMS settings.

**Comment by udf2457** [comment]
@hslatman I think you're missing the point. Let me try again....

The docs say _"put this in your ca.json"_:

```json
{
    "root": "/etc/step-ca/certs/root_ca.crt",
    "crt": "/etc/step-ca/certs/intermediate_ca.crt",
    "key": "awskms:key-id=f879f239-feb6-4596-9ed2-b1606277c7fe",
    "kms": {
        "type": "awskms",
        "uri": "awskms:region=us-east-2;profile=foo;credentials-file=/path/to/credentials"
    }
}
```

Meanwhile I invite you to [go read the reference](https://smallstep.com/docs/step-cli/reference/ca/init/#usage) for the `step ca init` command.

As I said. It is clear as mud as to how the CLI flags match up wth the ca.json stanzas.

**Comment by hslatman** [comment]
The `step ca init` doesn't have the flags for all configuration options, nor is it intended/designed to provide all of the options. It is provided as a way to get started quickly, and for further customization of the settings, we refer to our docs. You're free to build your own `ca.json` configuration with all bells and whistles applied based on the (partial) examples we provide in our docs. 

As mentioned, the `--remote-management` setting has _nothing_ to do with the AWS KMS settings.

**Comment by hslatman** [comment]
> The problem with the line of argument you're taking is [that the docs say otherwise](https://smallstep.com/docs/step-ca/provisioners/#remote-provisioner-management).

The docs also say [this](https://smallstep.com/docs/step-ca/provisioners/#on-an-existing-ca):

```On an Existing CA

....

```

> 
> I am told in the docs that if I want Remote Provisioner Management on a new CA, that the recommended way is to _"pass --remote-management to step ca init"_
> 
> But you're busy telling me `step ca init` is some sort of useless half-built tool and that if I want to do anything serious I like using AWS KMS in conjunction with Remote Provisioner Management then I should hack the `ca.json` manually.
> 
> It really makes no sense.

**Comment by udf2457** [comment]
Yeah ... "On an Existing CA".

My point is I do not have an existing CA.

Why should I be forced to play the "existing CA" game.

Is it not glaringly obvious that I should be able to set up a new CA, out of the box, with RPM and AWS KMS.

I mean come on, its not like I'm talking about some obscure use-case here.

**Comment by hslatman** [comment]
> Why should I be forced to play the "existing CA" game.

You're not forced to do anything, in my opinion. You're free to use and adapt the software. 
 
> Is it not glaringly obvious that I should be able to set up a new CA, out of the box, with RPM and AWS KMS ?

And you can do so, just not in the way you expect(ed) it to work out of the box / want it to work. Feel free to open a PR with improvements where you see fit.

**Comment by tashian** [comment]
The docs about how to set up AWS KMS are clear that you need to modify `ca.json`.
The docs for `step ca init` are clear that `--kms-root` only supports Azure right now.
It sounds like the software doesn't work the way you want it to work, but I don't see a docs issue here.
Closing.
